pkg/nodeadm/kubeadm.go

```diff
@@ -0,0 +1,114 @@
+package nodeadm
+
+import (
+	spv1 "github.com/platform9/ssh-provider/pkg/apis/sshprovider/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Networking contains elements describing cluster's networking configuration
+type Networking struct {
+	// ServiceSubnet is the subnet used by k8s services. Defaults to "10.96.0.0/12".
+	ServiceSubnet string `json:"serviceSubnet,omitempty"`
+	// PodSubnet is the subnet used by pods.
+	PodSubnet string `json:"podSubnet,omitempty"`
+	// DNSDomain is the dns domain used by k8s services. Defaults to "cluster.local".
+	DNSDomain string `json:"dnsDomain,omitempty"`
+}
+
+type KubeadmMasterConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	API                        API                  `json:"api,omitempty"`
+	APIServerCertSANs          []string             `json:"apiServerCertSANs,omitempty"`
+	Etcd                       Etcd                 `json:"etcd,omitempty"`
+	KubernetesVersion          string               `json:"kubernetesVersion,omitempty"`
+	Networking                 Networking           `json:"networking,omitempty"`
+	KubeletConfiguration       KubeletConfiguration `json:"kubeletConfiguration,omitempty"`
```

Comment:
Hi Daniel, just to understand, you'd mentioned that using upstream types adds noise in the serialized yaml. would that be an issue here as well.

Reply:
I agree. For this, I filed https://github.com/platform9/ssh-provider/issues/61 some time back.

Because ssh-provider does not introspect KubeletConfiguration or KubeProxyConfiguration, it might be reasonable to make both `map[string]interface{}` until the types are fixed upstream. I'll follow up with the above issue.